cmd/m3db-operator/main.go

```diff
@@ -83,35 +94,59 @@ func main() {
 		logger.Fatal("failed to create k8sclient", zap.Error(err))
 	}
 
+	stopCh := make(chan struct{})
+
+	// TODO(schallert): move these to k8sops client once we're confident we want
+	// them
+	//
+	// TODO(schallert): not sure if we need podinformers as well or if
+	// abstractions of statefulsets will do
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, _informerSyncDuration)
+	m3dbClusterInformerFactory := informers.NewSharedInformerFactory(crdClient, _informerSyncDuration)
+
 	// Create controller
-	controller, err := controller.New(logger, k8sclient)
+	controller, err := controller.New(logger, k8sclient, crdClient, kubeClient, kubeInformerFactory, m3dbClusterInformerFactory)
```

Comment:
should the controller have directly access to the `crdClient` and `kubeClient`?

Reply:
So far since I'm still changing stuff I've been using direct access to the clients rather than abstracting it behind something like k8sops. Once we have the full set of APIs / behaviors we depend on I think we can hide them.